crypto/bio/bio_lib.c

```diff
@@ -502,7 +502,7 @@ int BIO_get_line(BIO *bio, char *buf, int size)
         return -1;
     }
 
-    while (size-- > 1 && (ret = BIO_read(bio, ptr, 1)) > 0)
+    while (size-- > 1 && !BIO_eof(bio) && (ret = BIO_read(bio, ptr, 1)) > 0)
```

Comment:
I am wondering if BIO_eof() is always set immediately after the last possible read. I.E., if this will work properly on the case of no EOL at EOF.

It might be better to just check the BIO_eof() at the return. `return ret > 0 || BIO_eof(bio) ? ptr - buf : ret;`

Reply:
Good thought - checking `BIO_eof()` after the loop should be safer, and it's definitely more efficient!
Done.